rust/routee-compass/src/app/compass/config/traversal_model/energy_model_vehicle_builders.rs

```diff
@@ -0,0 +1,125 @@
+use std::sync::Arc;
+
+use routee_compass_core::util::unit::{
+    Energy, EnergyRate, EnergyRateUnit, EnergyUnit, GradeUnit, SpeedUnit,
+};
+use routee_compass_powertrain::routee::{
+    prediction::{load_prediction_model, model_type::ModelType, PredictionModelRecord},
+    vehicles::{
+        default::{conventional::ConventionalVehicle, plug_in_hybrid::PlugInHybrid},
+        Vehicle,
+    },
+};
+
+use crate::app::compass::config::{
+    compass_configuration_error::CompassConfigurationError,
+    compass_configuration_field::CompassConfigurationField,
+    config_json_extension::ConfigJsonExtensions,
+};
+
+pub trait VehicleBuilder {
```

Comment:
Yeah good point, conventional is misleading since it can be any powertrain type that just reports a single fuel. Something tricky here is that the hybrid models also just return fuel in gasoline and are suitable for this "conventional" model. The "plugin hybrid" is just a special case for a PHEV vehicle. So, we have a weird setup where we need to find a name for the model that represents the set {ICE, BEV, HEV} and the set {PHEV}. Maybe something like "SingleFuelVehicle" and "DualFuelVehicle" that represents the fuel that goes _into_ the vehicle?

Reply:
> Something tricky here is that the hybrid models also just return fuel in gasoline and are suitable for this "conventional" model

but, is that an implementation detail of how we're currently talking about hybrid, or an implementation detail of how FASTSim does it? i thought our long-term goal might be that these vehicles can refuel. if we don't want to re-write everything for that behavior, then, i would assume hybrid would be a special case of the PHEV code where there simply is no refuel behavior.

as for names, something like SingleFuelVehicleModel and HybridFuelVehicleModel might be general enough?

Reply:
Something that the above leaves out is that there will still be shared behavior between some of the models. To address that, maybe we can just centralize the shared behavior into functions that are shared between the different vehicles.